test/integration/node-specific/client_close.test.ts

```diff
@@ -128,25 +329,138 @@ describe.skip('MongoClient.close() Integration', () => {
       describe('ConnectionPool', () => {
         describe('Node.js resource: minPoolSize timer', () => {
           describe('after new connection pool is created', () => {
-            it.skip('the minPoolSize timer is cleaned up by client.close()', async () => {});
+            it('the minPoolSize timer is cleaned up by client.close()', async function () {
+              const run = async function ({ MongoClient, uri, expect, getTimerCount }) {
+                const client = new MongoClient(uri, { minPoolSize: 1 });
+                let minPoolSizeTimerCreated = false;
+                client.on('connectionPoolReady', () => (minPoolSizeTimerCreated = true));
+                await client.connect();
+
+                expect(minPoolSizeTimerCreated).to.be.true;
+
+                const servers = client.topology?.s.servers;
+
+                function getMinPoolSizeTimer(servers) {
+                  for (const server of servers) {
+                    return server[1].pool.minPoolSizeTimer;
+                  }
+                }
+                // note: minPoolSizeCheckFrequencyMS = 100 ms by client, so this test has a chance of being flaky
+                expect(getMinPoolSizeTimer(servers)).to.exist;
+
+                await client.close();
+                expect(getMinPoolSizeTimer(servers)).to.not.exist;
+                expect(getTimerCount()).to.equal(0);
+              };
+              await runScriptAndGetProcessInfo('timer-min-pool-size', config, run);
+            });
           });
         });
 
         describe('Node.js resource: checkOut Timer', () => {
-          // waitQueueTimeoutMS
           describe('after new connection pool is created', () => {
-            it.skip('the wait queue timer is cleaned up by client.close()', async () => {});
+            it('the wait queue timer is cleaned up by client.close()', async function () {
+              const run = async function ({
+                MongoClient,
+                uri,
+                expect,
+                sinon,
+                sleep,
+                getTimerCount,
+                timers
+              }) {
+                const waitQueueTimeoutMS = 1515;
+
+                // configure failPoint
+                const utilClient = new MongoClient(uri);
+                await utilClient.connect();
+                const failPoint = {
+                  configureFailPoint: 'failCommand',
+                  mode: { times: 1 },
+                  data: {
+                    blockConnection: true,
+                    blockTimeMS: waitQueueTimeoutMS * 3,
+                    failCommands: ['insert']
+                  }
+                };
+                await utilClient.db('admin').command(failPoint);
+
+                const timeoutStartedSpy = sinon.spy(timers, 'setTimeout');
```

Comment:
I'm seeing an error thrown for this the same way it was for the server selection test. I don't think we need the spy if we know we have a commandStarted on a client with maxPoolSize: 1

Reply:
Sounds good, replaced the spy with the awaiting a commandStarted event